server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java

```diff
@@ -263,6 +278,224 @@ TransportResponse handleExtensionRequest(ExtensionRequest extensionRequest) {
         return null;
     }
 
+    public void getNamedWriteables() {
+
+        // retrieve named writeable registry entries from each extension
+        for (DiscoveryNode extensionNode : extensionsList) {
+            try {
+                Map<DiscoveryNode, Map<Class<?>, Map<String, ExtensionReader>>> extensionRegistry = getNamedWriteables(extensionNode);
+                if (extensionRegistry.isEmpty() == false) {
+                    this.extensionNamedWriteableRegistry.putAll(extensionRegistry);
+                }
+            } catch (UnknownHostException e) {
+                logger.error(e.toString());
+            }
+        }
+
+        // TODO : enable dynamic registration of named writeables by returning the extension registry map within Node.java
+    }
+
+    public Map<DiscoveryNode, Map<Class<?>, Map<String, ExtensionReader>>> getNamedWriteables(DiscoveryNode extensionNode)
+        throws UnknownHostException {
+
+        final CountDownLatch inProgressLatch = new CountDownLatch(1);
+
+        // initialize map of entries for this extension to return
+        final Map<DiscoveryNode, Map<Class<?>, Map<String, ExtensionReader>>> extensionRegistry = new HashMap<>();
+
+        final TransportResponseHandler<NamedWriteableRegistryResponse> namedWriteableRegistryResponseHandler = new TransportResponseHandler<
+            NamedWriteableRegistryResponse>() {
+
+            @Override
+            public NamedWriteableRegistryResponse read(StreamInput in) throws IOException {
+                return new NamedWriteableRegistryResponse(in);
+            }
+
+            @Override
+            public void handleResponse(NamedWriteableRegistryResponse response) {
+
+                logger.info("response {}", response);
+
+                logger.info("EXTENSION [" + extensionNode.getName() + "] returned " + response.getRegistry().size() + " entries");
+                if (response.getRegistry().isEmpty() == false) {
+
+                    // Extension has sent over entries to register, initialize inner category map
+                    Map<Class<?>, Map<String, ExtensionReader>> categoryMap = new HashMap<>();
```

Comment:
What's the need to have Class type as wildcard and not specific here and the for the rest?